src/codegen/compile.ml

```diff
@@ -9459,20 +9472,31 @@ and compile_dec env pre_ae how v2en dec : VarEnv.t * G.t * (VarEnv.t -> scope_wr
     assert AllocHow.(match M.find_opt name how with
                      | Some (LocalMut _ | StoreHeap | StoreStatic) -> true
                      | _ -> false);
-      let pre_ae1, alloc_code = AllocHow.add_local env pre_ae how name in
+    let pre_ae1, alloc_code = AllocHow.add_local env pre_ae how name in
 
-      ( pre_ae1,
-        alloc_code,
-        (fun ae -> let sr, code = Var.set_val env ae name in compile_exp_as env ae sr e ^^ code),
-        unmodified
-      )
+    ( pre_ae1,
+      alloc_code,
+      (fun ae -> let sr, code = Var.set_val env ae name in compile_exp_as env ae sr e ^^ code),
+      unmodified
+    )
+  | RefD (name, _, e) ->
+    let pre_ae1, _ = AllocHow.add_local env pre_ae how name in
```

Comment:
Once this works I'd rather not use a function and throw away a result, but just do whats need. Maybe add a `add_local_for_alias` or something. Or maybe there is already a function that does exactly what's needed here.

Reply:
404c24fd8!